### PR TITLE
groups/data.js translation to TS

### LINKED ISSUE #10: Translate groups/data.js file
Issue 1D

- Estimate Effort: 3 days
- Dependencies: None
- Initial Assignments: @kateyzcodes 
- Milestones: Completed in Sprint 1
- Acceptance Criteria:
  - Passes the linter
  - Passes the tests
  - Builds properly

### PR DESCRIPTION
**Pull request by kateyzcodes**
There is a timeout error relating to the .create() function in Groups.

### CODE REVIEW

**Comment by cwesterink** [comment]
Had to edit one of the test files to stop some of the tests from failing

File: src/groups/data.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        const results = await plugins.hooks.fire('filter:groups.get', { groups: groupData }) as { groups: GroupDataObject[] };
+        return results.groups;
+    };
+
+    Groups.getGroupsData = async function (groupNames: string[]): Promise<GroupDataObject[]> {
+        return await Groups.getGroupsFields(groupNames, []);
+    };
+
+    Groups.getGroupData = async function (groupName: string): Promise<GroupDataObject | null> {
+        console.log('getd1');
```
Comment (Sebastian0011) [inline src/groups/data.ts]: Remove console logs

File: src/groups/data.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        const groupsData = await Groups.getGroupsData([groupName]);
+        console.log('d2');
```
Comment (Sebastian0011) [inline src/groups/data.ts]: Remove console logs

File: src/groups/data.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        return groupData ? groupData[field] as GroupDataField : null;
+    };
+
+    Groups.getGroupFields = async function (groupName: string, fields: string[]): Promise<GroupDataObject | null> {
+        const groups = await Groups.getGroupsFields([groupName], fields);
+        return groups ? groups[0] : null;
+    };
+
+    Groups.setGroupField = async function (groupName: string, field: string, value: GroupDataField): Promise<void> {
+        console.log('checp1');
```
Comment (Sebastian0011) [inline src/groups/data.ts]: Remove console logs

File: src/groups/data.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        await db.setObjectField(`group:${groupName}`, field, value);
+        console.log('checp2');
+        await plugins.hooks.fire('action:group.set', { field: field, value: value, type: 'set' });
```
Comment (Sebastian0011) [inline src/groups/data.ts:125]: Does this need to be ignored in ESLint?

**Review by Sebastian0011 — CHANGES_REQUESTED** [review]
Sebastian's Review

**Review by Sebastian0011 — APPROVED** [review]

**Pull request merged**